charts/partials/templates/_proxy.tpl

```diff
@@ -0,0 +1,83 @@
+{{ define "partials.proxy" -}}
+- env:
+  - name: LINKERD2_PROXY_LOG
+    value: {{.LogLevel}}
+  - name: LINKERD2_PROXY_DESTINATION_SVC_ADDR
+    value: {{ternary "localhost.:8086" (printf "linkerd-destination.%s.svc.%s:8086" .ControlPlaneNamespace .ClusterDomain) (eq .Component "controller")}}
+  - name: LINKERD2_PROXY_CONTROL_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Control}}
+  - name: LINKERD2_PROXY_ADMIN_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Admin}}
+  - name: LINKERD2_PROXY_OUTBOUND_LISTEN_ADDR
+    value: 127.0.0.1:{{.Port.Outbound}}
+  - name: LINKERD2_PROXY_INBOUND_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Inbound}}
+  - name: LINKERD2_PROXY_DESTINATION_PROFILE_SUFFIXES
+    {{- $internalProfileSuffix := printf "svc.%s" .ClusterDomain }}
+    value: {{ternary "." $internalProfileSuffix .EnableExternalProfile}}
+  - name: LINKERD2_PROXY_INBOUND_ACCEPT_KEEPALIVE
+    value: {{.InboundAcceptKeepAlive}}
+  - name: LINKERD2_PROXY_OUTBOUND_CONNECT_KEEPALIVE
+    value: {{.OutboundAcceptKeepAlive}}
+  - name: _pod_ns
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: metadata.namespace
+  - name: LINKERD2_PROXY_DESTINATION_CONTEXT
+    value: ns:$(_pod_ns)
```

Comment:
Can you please include this to account for the Prometheus special case?
```
  {{ - if eq .Component "prometheus" -}}
  - name: LINKERD2_PROXY_OUTBOUND_ROUTER_CAPACITY
    value: 10000
  {{- end }}
```

Reply:
Also all the following env vars should be wrapped around this:
```
{{- if eq .DisableIdentity true -}}
- name: LINKERD2_PROXY_IDENTITY_DISABLED
  value: disabled
{{ else }}
....
{{- end }}
```

Reply:
I found another one to add here:
```
  {{ - if eq .DisableTap true -}}
  - name: LINKERD2_PROXY_TAP_DISABLED
    value: "true"
  {{- end }}
```

Reply:
Come to think of it, I don't think we should add `DisableIdentity` and `DisableTap` to the proxy partial, because they will allow users to disable identity and tap in the control plane during install. I don't think `helm install` should introduce behaviours that are supported by `linkerd install`. These inject-specific behaviour should go in the `inject` package.

Reply:
Agreed on not exposing that to the install workflow. In order to still being able to use the same partial in both scenarios, what do you think about adding a condition such as this?
```
 {{ - if and (ne .Chart.name "linkerd") .DisableTap -}}
  - name: LINKERD2_PROXY_TAP_DISABLED
    value: "true"
  {{- end }}
```

Reply:
I wonder if we can let the `inject` package handle the inject-specific logic, like what it's already doing today with disabling tap, disabling identity, copying primary container's security context, mounting service account token volume etc. (This will mean we need to marshal and unmarshal the partials into Go structs.) I think conditions relying on `.Chart.name` can become hard to track over time.

Alternately, can we introduce a variable in the `partials` chart, that is only set to `true` by the `values` struct in the `inject` package (like how we create the `installValues` in `install.go`)?

If we fancy, we can also try [tags and conditions](https://helm.sh/docs/developing_charts/#tags-and-condition-fields-in-requirements-yaml) to control whether the `partials` chart will include a (new) `partials-inject` chart.

```
# partials/requirements.yaml
dependencies:
- name: partials-inject
  repository: file://../partials-inject
  version: 0.1.0
  condition: partials-inject.enabled

# set to `true` only in `values` struct in the `inject` package
# otherwise, always false
partials-inject:
  enabled: true

# partials/_proxy.tpl
{{ - if and .partials-inject.enabled .DisableTap -}}
{{- include "partials-inject.disabled-tap -}}
{{- else -}}
# whatever defaults
{{- end }}
...
{{ - if and .partials-inject.enabled .DisableIdentity -}}
{{- include "partials-inject.disabled-identity -}}
{{- else -}}
# whatever defaults
{{- end }}
...
{{ - if .partials-inject.enabled -}}
{{- include "partials-inject.volume-mount.service-account -}}
{{- else -}}
# whatever defaults
{{- end }}
```

Reply:
Also, I *think* `.Chart.name` will always be `partials`, since it's just reading off whatever is defined in the `Chart.yaml`.

Reply:
I'd personally rather keep the surface area minimal and provide an advanced solution with all the knobs than create some special logic that depends on where things are being used. If someone *really* wants to disable tap/tls in the control plane then we should let them.

Reply:
Per post stand-up discussion, for simplicity, I added the `DisableIdentity` and `DisableTap` variables to the `values.yaml`  and some checks in the templates to ensure identity and tap are disabled on control plane components.

Reply:
Also added the new `LINKERD2_PROXY_TAP_SVC_NAME` env var per #3155.